### PR TITLE
[AUTO] Set  property optimal_number_of_infer_requests to hw device with default value

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - Query the property optimal_number_of_infer_requests from plugin first if plugin supported.
 - calculate the optimal_number_of_infer_requests if no default value from hw device

### Tickets:
 - 110248

### CODE REVIEW

File: src/plugins/auto/src/auto_executable_network.cpp [hunk 1]
```diff
@@ -145,9 +145,20 @@ IE::Parameter AutoExecutableNetwork::GetMetric(const std::string& name) const {
...
+            real = _autoSchedule->_loadContext[ACTUALDEVICE].executableNetwork->GetMetric(name).as<unsigned int>();
         } else {
+            auto supportedProperties =
+                _autoSContext->_core->get_property(_autoSchedule->_loadContext[ACTUALDEVICE].deviceInfo.deviceName,
+                                                   ov::supported_properties);
+            auto nireqIter =
+                std::find(supportedProperties.begin(), supportedProperties.end(), ov::optimal_number_of_infer_requests);
+            if (nireqIter != supportedProperties.end()) {
+                real =
+                    _autoSContext->_core->get_property(_autoSchedule->_loadContext[ACTUALDEVICE].deviceInfo.deviceName,
```
Comment (peterchen-intel) [inline src/plugins/auto/src/auto_executable_network.cpp:157]: Need to pass the properties which are to be passed to compile_model

Comment (peterchen-intel) [reply src/plugins/auto/src/auto_executable_network.cpp:157]: Please add test cases.

Comment (yangwang201911) [reply src/plugins/auto/src/auto_executable_network.cpp]: Updated.

File: src/plugins/auto/src/auto_compiled_model.cpp [hunk 2]
```diff
@@ -109,81 +109,92 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
...
-                        m_context->m_ov_core->get_property(device_info.device_name,
-                            ov::hint::performance_mode)== ov::hint::PerformanceMode::THROUGHPUT;
-            } catch (const ov::Exception&) {
-                LOG_DEBUG_TAG("get_property:%s for %s", "PERF_HINT config not supported",
-                    device_info.device_name.c_str());
+            auto actual_dev_supported_properties =
+                m_context->m_ov_core->get_property(device_info.device_name, ov::supported_properties);
+            auto ireqIter = std::find(actual_dev_supported_properties.begin(), actual_dev_supported_properties.end(), name);
+            if (ireqIter != actual_dev_supported_properties.end()) {
+                real = m_context->m_ov_core->get_property(device_info.device_name, ov::optimal_number_of_infer_requests);
```
Comment (songbell) [inline src/plugins/auto/src/auto_compiled_model.cpp:122]: are we sure that this is the vpux plugin supported metric? in cpu/gpu, this seems to be a compiled model metric.
also, how do we place this ov::optimal_number_of_infer_requests against ov::num_request?

Comment (peterchen-intel) [inline src/plugins/auto/src/auto_compiled_model.cpp:122]: Need pass all configurations to plugin. 3rd parameter of get_property()

Comment (peterchen-intel) [reply src/plugins/auto/src/auto_compiled_model.cpp:122]: 1. ov::num_request pass to all devices in CTPUT hint - will have multiple times optimal_number_of_infer_requests
2. Without CPU as acceleration, get from compiled model
3. With CPU as acceleration
a. Get from plugin
pass all configurations when getting optimal_number_of_infer_requests from HW device
https://docs.openvino.ai/2022.3/classov_1_1Core.html#doxid-classov-1-1-core-1a3267437192896805958f2d6ccfd66429
b. Estimate by AUTO
min(ov::num_request, optimal_number_of_infer_requests estimated  by AUTO)

Comment (yangwang201911) [reply src/plugins/auto/src/auto_compiled_model.cpp:126]: Updated.

Comment (yangwang201911) [reply src/plugins/auto/src/auto_compiled_model.cpp]: Updated.

File: src/plugins/auto/src/auto_compiled_model.cpp [hunk 3]
```diff
@@ -108,50 +108,76 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
...
+                                                      ov::hint::num_requests) != actual_dev_supported_properties.end();
+            auto reqs_iter = device_info.config.find(ov::hint::num_requests.name());
+            auto ireq_iter = std::find(actual_dev_supported_properties.begin(), actual_dev_supported_properties.end(), name);
+            if (ireq_iter != actual_dev_supported_properties.end()) {
+                real = m_context->m_ov_core->get_property(device_info.device_name,
+                                                          ov::optimal_number_of_infer_requests,
+                                                          device_info.config);
+            }
+            if (real > 0) {
+                if (is_supported_num_request && reqs_iter != device_info.config.end()) {
```
[the remark this answers is not in the document]
Comment (yangwang201911) [reply src/plugins/auto/src/auto_compiled_model.cpp:130]: Following provided test cases by @wangleis , will consider if `ov::num_requests` is supported by hardware in this case. Is there any update for the listed test cases in the JIRA?

File: src/plugins/auto/src/auto_compiled_model.cpp [hunk 4]
```diff
@@ -108,50 +108,76 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
...
+                    requests = reqs_iter->second.as<unsigned int>();
+                    real = (std::min)(requests, real);
```
Comment (songbell) [inline src/plugins/auto/src/auto_compiled_model.cpp:132]:
```suggestion
                    if (requests > 0)
                        real = (std::min)(requests, real);
```

Comment (songbell) [reply src/plugins/auto/src/auto_compiled_model.cpp:132]: just want to make sure that the requests num here is not some default value carried from auto

File: src/plugins/auto/src/auto_compiled_model.cpp [hunk 5]
```diff
@@ -184,6 +208,10 @@ ov::Any AutoCompiledModel::get_property(const std::string& name) const {
             } else {
                 real = default_num_for_latency;
             }
+            if (reqs_iter != device_info.config.end()) {
+                requests = reqs_iter->second.as<unsigned int>();
```
Comment (songbell) [inline src/plugins/auto/src/auto_compiled_model.cpp:212]: same here

File: src/plugins/auto/src/auto_compiled_model.cpp (hunk 4 above)
Comment (yangwang201911) [reply src/plugins/auto/src/auto_compiled_model.cpp:132]: Sorry, didn't understand clearly what you meant. Could you please help provide a test case for this scenario?

File: src/plugins/auto/src/auto_compiled_model.cpp (hunk 5 above)
Comment (yangwang201911) [reply src/plugins/auto/src/auto_compiled_model.cpp:212]: Updated.

**Review by songbell — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

**Pull request merged**